### PR TITLE
tail as an organ

### PR DESCRIPTION
**Pull request by SolatK**
## Description of changes
new organ - tail

## Authorship
Me

## Changelog

:cl:
rscadd: The tail is now a separate limb. You can select it as target, and button is shown when you look at somebody with a tail.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -678,93 +678,38 @@ var/global/list/damage_icon_parts = list()
...
+		var/icon/tail_icon = get_tail_icon(tail_organ)
+		overlays_standing[(dir == NORTH) ? HO_TAIL_OVER_LAYER : HO_TAIL_UNDER_LAYER] = image(tail_icon)
 
 	if(update_icons)
 		update_icon()
 
-/mob/living/carbon/human/proc/get_tail_icon()
-	var/icon_key = "[bodytype.get_icon_cache_uid(src)][skin_colour][hair_colour]"
+/mob/living/carbon/human/proc/get_tail_icon(var/obj/item/organ/external/tail/tail_organ)
+	var/icon_key = "[bodytype.get_icon_cache_uid(src)][skin_colour][hair_colour][bodytype.name]"
```
Comment (MistakeNot4892) [inline code/modules/mob/living/carbon/human/update_icons.dm]: What's the purpose of adding `bodytype.name`? `get_icon_cache_uid()` should already guarantee the bodytype is unique.

File: code/__defines/items_clothing.dm [hunk 2]
```diff
@@ -186,9 +186,10 @@
 #define BP_HEAD         "head"
 #define BP_CHEST        "chest"
 #define BP_GROIN        "groin"
+#define BP_TAIL         "tail"
 
-var/global/list/all_limb_tags = list(BP_CHEST, BP_GROIN, BP_HEAD, BP_L_ARM, BP_R_ARM, BP_L_HAND, BP_R_HAND, BP_L_LEG, BP_R_LEG, BP_L_FOOT, BP_R_FOOT)
-var/global/list/all_limb_tags_by_depth = list(BP_HEAD, BP_L_HAND, BP_R_HAND, BP_L_ARM, BP_R_ARM, BP_L_FOOT, BP_R_FOOT, BP_L_LEG, BP_R_LEG, BP_GROIN, BP_CHEST)
+var/global/list/all_limb_tags = list(BP_CHEST, BP_GROIN,BP_TAIL , BP_HEAD, BP_L_ARM, BP_R_ARM, BP_L_HAND, BP_R_HAND, BP_L_LEG, BP_R_LEG, BP_L_FOOT, BP_R_FOOT)
```
Comment (MistakeNot4892) [inline code/__defines/items_clothing.dm]: Random space snuck in.

File: code/modules/organs/external/tail.dm [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	parent_organ = BP_GROIN
+	joint = "tail"
+	amputation_point = "tail"
+	artery_name = "vein"
+	arterial_bleed_severity = 03
+	limb_flags = ORGAN_FLAG_CAN_AMPUTATE | ORGAN_FLAG_CAN_BREAK
+
+/obj/item/organ/external/tail/set_dna(datum/dna/new_dna)
+	. = ..()
+	if(owner.bodytype || !(tail_icon && tail_state && tail_hair && tail_animation))//no overriding if something there
```
Comment (MistakeNot4892) [inline code/modules/organs/external/tail.dm]: Why do you store these values on the tail anyway? Why not just grab them from `owner.bodytype` when you are generating the icon?

File: code/modules/species/species_bodytype_helpers.dm [hunk 4]
```diff
@@ -31,10 +31,17 @@
 	return get_deform ? icon_deformed : icon_base
 
 /decl/bodytype/proc/get_tail(var/mob/living/carbon/human/H)
-	return tail
+	var/obj/item/organ/external/tail/tail = H.organs_by_name[BP_TAIL]
```
Comment (MistakeNot4892) [inline code/modules/species/species_bodytype_helpers.dm:34]: Using bodytype helper procs at all seems strange if tails are being moved entirely to an organ - you might as well just make the tail appearance a property of the tail organ specifically and use tail subtypes to customize it per-species.

File: mods/species/lizard/datum/species.dm [hunk 5]
```diff
@@ -89,6 +89,21 @@
 		BP_BRAIN = /obj/item/organ/internal/brain/lizard
 	)
 
+	has_limbs = list(
```
Comment (MistakeNot4892) [inline mods/species/lizard/datum/species.dm]: There's another list you can use to override specific limb flags:
````
override_limb_types = list(BP_HEAD = /obj/item/organ/external/head/no_eyes)
````

File: code/modules/organs/external/tail.dm (hunk 3 above)
Comment (SolatK) [reply code/modules/organs/external/tail.dm]: The tail should not adjust to the body during surgery. So, data on its appearance should be saved in it.

File: code/modules/species/species_bodytype_helpers.dm (hunk 4 above)
Comment (SolatK) [reply code/modules/species/species_bodytype_helpers.dm:34]: Yes, I need to move that to the organ

**Review by MistakeNot4892 — DISMISSED** [review]
There are some parts of this I would like to tidy up/change after this is merged (specifically moving tail info entirely onto the tail organ instead of bodytype) but this is fine as is.

**Review by MistakeNot4892 — DISMISSED** [review]

File: code/modules/organs/external/tail.dm [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	name = "tail"
+	max_damage = 20
+	min_broken_damage = 10
+	w_class = ITEM_SIZE_NORMAL
+	body_part = SLOT_LOWER_BODY
+	parent_organ = BP_GROIN
+	joint = "tail"
+	amputation_point = "tail"
+	artery_name = "vein"
+	arterial_bleed_severity = 03
```
Comment (comma) [inline code/modules/organs/external/tail.dm]: 03? did you mean 3 or 0 or 0.3?

Comment (SolatK) [reply code/modules/organs/external/tail.dm]: 0.3

**Review by comma — APPROVED** [review]

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**